### PR TITLE
Ms2 interactive list

### PR DESCRIPTION
**Pull request by Bishoy-Samwel**
- [x] Add 2 new JavaScript files and import them as modules:

>*  one will contain methods related to drag / drop  and sorting functionality.
>*  the other will contain methods related to the status updates (completed: true / false).

- [x]  Add event listeners

>*  to list items (dragstart, dragover, drop) and
>*  to checkbox (change).

- [x] Update items object's values for     

> *  index and    
> *  completed keys upon user actions.

- [x] Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: src/list.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        this.tasks.unshift(current);
+      }
+    } else {
+      // top to down edge
+      this.tasks.push(current);
+    }
+    this.tasks.forEach((task) => {
+      task.index = this.tasks.indexOf(task);
+    });
+    localStorage.setItem('tasks', JSON.stringify(this.tasks));
```
Comment (vichuge) [inline src/list.js:47]: - [x] Talking about `localStorage`, I tried to check all elements in the list and then I refresh the page. It looks like you have a bug with local storage and the first element. I kindly invite you to fix it. 🐶 
image

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 
You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your README and PR message are descriptive ✔️

## Aspects to improve ♻️

- [x] This is not a point for this project but is important too. I'm not able to see the names for each element in the list, and for this reason, if I change the element position I couldn't know if I change it or not. It looks like I can drag and drop elements, but I'm not able to see if the elements remain in their new positions if I refresh because all of them have `undefined` as a title. Please, fix this bug for titles and check `localStorage` is working well when you drag/drop an element and then refresh the page.
image


- [x] Check the comments under the review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/list.js (hunk 1 above)
Comment (Bishoy-Samwel) [reply src/list.js:47]: I cleared `localStorage` data and It works fine on my side.
image

Comment (Bishoy-Samwel) [reply src/list.js:47]: Maybe you need to clear `localStorage` first on your side because I initialized the list object with an array of 5 tasks, not 3.

**Review by mimipeshy — APPROVED** [review]
## Hi @Bishoy-Samwel 👋🏾 👋🏾 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](<https://microverse.zendesk.com/hc/en-us/articles/1500004088561>)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](<https://airtable.com/shrQAqnBwek5a0O0s>)._

**Pull request merged**